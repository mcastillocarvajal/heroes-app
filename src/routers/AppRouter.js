import React, { useContext } from 'react'
import {
    BrowserRouter as Router,
    Switch,
  } from "react-router-dom";
import { AuthContext } from '../auth/AuthContext';
import { PublicRoute } from './PublicRoute';
import { PrivateRoute } from './PrivateRoute';
import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';



export const AppRouter = () => {

    const { user } = useContext( AuthContext );


    return (
        <Router>
            <div>

                <Switch>
                    <PublicRoute exact path="/login" component={ LoginScreen } isAuth={ user.logged }/>

                    <PrivateRoute path="/" component={ DashboardRoutes } isAuth={ user.logged } />
                </Switch>

            </div>
        </Router>
    );
}
