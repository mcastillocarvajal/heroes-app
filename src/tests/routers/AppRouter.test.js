import React, { PureComponent } from 'react';
import { mount } from 'enzyme';
import { AppRouter } from '../../routers/AppRouter';
import { AuthContext } from '../../auth/AuthContext';

describe('test for <AppRouter />', () => {

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    }
    
    test('should show login if user is not auth', () => {
        
        const wrapper = mount ( 
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect( wrapper ).toMatchSnapshot();
    })

    test('should show marvel component if user is auth', () => {

        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: true,
                name: 'Moises'
            }
        }

        const wrapper = mount ( 
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect( wrapper.find('.navbar').exists() ).toBe(true);
    })
    
    
})
