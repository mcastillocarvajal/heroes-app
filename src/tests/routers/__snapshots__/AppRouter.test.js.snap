// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for <AppRouter /> should show login if user is not auth 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
       Login Screen
    </h1>
    <hr />
    <button
      className="btn btn-dark"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
